shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -134,44 +134,45 @@ - (void)userNotificationCenter:(UNUserNotificationCenter*)center
   }
 }
 
-- (BOOL)openURL:(NSURL*)url {
+- (BOOL)isFlutterDeepLinkingEnabled {
   NSNumber* isDeepLinkingEnabled =
       [[NSBundle mainBundle] objectForInfoDictionaryKey:@"FlutterDeepLinkingEnabled"];
-  if (!isDeepLinkingEnabled.boolValue) {
-    // Not set or NO.
-    return NO;
-  } else {
-    FlutterViewController* flutterViewController = [self rootFlutterViewController];
-    if (flutterViewController) {
-      [flutterViewController.engine
-          waitForFirstFrame:3.0
-                   callback:^(BOOL didTimeout) {
-                     if (didTimeout) {
-                       FML_LOG(ERROR)
-                           << "Timeout waiting for the first frame when launching an URL.";
-                     } else {
-                       [flutterViewController.engine.navigationChannel
-                           invokeMethod:@"pushRouteInformation"
-                              arguments:@{
-                                @"location" : url.absoluteString ?: [NSNull null],
-                              }];
-                     }
-                   }];
-      return YES;
-    } else {
-      FML_LOG(ERROR) << "Attempting to open an URL without a Flutter RootViewController.";
-      return NO;
-    }
-  }
+  // if not set, return NO
+  return isDeepLinkingEnabled ? [isDeepLinkingEnabled boolValue] : NO;
 }
 
+// This method is called when opening an URL with custom schemes.
 - (BOOL)application:(UIApplication*)application
             openURL:(NSURL*)url
             options:(NSDictionary<UIApplicationOpenURLOptionsKey, id>*)options {
   if ([_lifeCycleDelegate application:application openURL:url options:options]) {
     return YES;
   }
-  return [self openURL:url];
+  return [self handleOpenURL:url options:options throwBackToiOS:NO];
+}
+
+// Helper function for opening an URL.
+- (BOOL)handleOpenURL:(NSURL*)url
+              options:(NSDictionary<UIApplicationOpenURLOptionsKey, id>*)options
+       throwBackToiOS:(BOOL)throwBack {
```

Comment:
nit: `throwBackToiOS` -> `relayToSystemIfUnhandled`. Can also add a brief comment describe this bool.